### ISSUE TITLE
Media-Upload with S3 Adapter not working

### ISSUE THREAD

**Issue by sschwarz-allbuyone** [shopware/platform#2889]
### PHP Version

8.1.13

### Shopware Version

6.4.18

### Expected behaviour

A Upload of an image via Media-Manager in the administration should upload the file to a Amazon-S3 Bucket or at least should give an error-message, that the upload is failed.

### Actual behaviour

When i try to upload a image via content/media there is the success message “File has been saved” in the administration, but there is no file uploaded to s3 and therefore no image to display. There also no file in the s3 bucket. With Version 6.4.17 the upload is working like expected with the same config and bucket.

### How to reproduce

Place the shopware.yaml in the config/packages/ Folder. Go in the Administration an try to Upload a Image. 

```
shopware:
  filesystem:
    public:
      url: "https://%env(AWS_DOMAIN)%"
      type: "amazon-s3"
      visibility: "public"
      config:
        bucket: "%env(AWS_BUCKET)%"
        region: "eu-west-1"
        credentials:
          key: "%env(AWS_ACCESS_KEY_ID)%"
          secret: "%env(AWS_SECRET_ACCESS_KEY)%"
        endpoint: "https://s3.eu-west-1.amazonaws.com/%env(AWS_BUCKET)%"
        root: "/"
```

**Comment by shyim** [comment]
Can you try
```
FEATURE_NEXT_14699=1
```

**Comment by sschwarz-allbuyone** [comment]
@shyim thx for your response. How do i use the flag FEATURE_NEXT_14699=1 properly?
I tried it in my .env and cleared the cache but it didn't have any effect.

**Comment by shyim** [comment]
The fix is on way with S3. Something changed in the S3 Adapter itself from AWS

**Comment by reiterth** [comment]
Same problem here, but i get "Error while uploading"
I tried with 6.4.17.2 and 6.4.18.0

In Browser network debug i can see:

```
PutObject\u0022 on \u0022http:\/\/xxxxxx.s3.eu-central-1.amazonaws.com\/xxxxxx\/media\/c5\/9a\/4a\/1674664412\/istockphoto-1400082088-612x612.jpg\u0022; AWS HTTP error: Client error: PUT http:\/\/xxxxxx.s3.eu-central-1.amazonaws.com\/xxxxxx\/media\/c5\/9a\/4a\/1674664412\/istockphoto-1400082088-612x612.jpg resulted in a 403 Forbidden
```

Upload with the same credentials with  winSCP works perfect!

Here my shopware.yaml

```
shopware:
    filesystem:
        public:
            url: "https://%env(AWS_BUCKET)%.s3.eu-central-1.amazonaws.com"
            type: "amazon-s3"
            visibility: "public"
            config:
                bucket: "%env(AWS_BUCKET)%"
                region: "eu-central-1"
                credentials:
                    key: "%env(AWS_ACCESS_KEY_ID)%"
                    secret: "%env(AWS_SECRET_ACCESS_KEY)%"
                endpoint: "https://s3.eu-central-1.amazonaws.com"
                root: "/"
                options:
                    visibility: "public"


```

.env

![image](https://user-images.githubusercontent.com/51745968/214624961-3dbc9133-6223-4e8c-8c98-2d85c59a5423.png)

**Comment by shyim** [comment]
It is fixed with 6.4.18.1 or run `composer require aws/aws-sdk-php 3.242`

**Comment by sschwarz-allbuyone** [comment]
@shyim: Are you sure, that the fix is in Shopware 6.4.18.1? It didn't work for me. But composer require aws/aws-sdk-php 3.242 did the job.
